From c3f028f9df6eebe7d5f73276c73db9e4c525153d Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Fri, 14 Jun 2019 10:10:28 +0800
Subject: [PATCH] arm64: dts: rockchip: reasonable alllcation of VOP on rk3399
 linux

Most users used 4K monitors with HDMI on RK3399/RK3399PRO platform, but no
more than 2K monitors on EDP/DSI panels.

So the reasonable that HDMI connected to VOPB control, and EDP/DSI shound
be connected to VOPL.

Change-Id: Id97efc5d6d534c302aa52ad00e705c093457f41e
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi            | 6 +++---
 .../boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts | 8 ++++++++
 arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts  | 4 ++--
 arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts  | 4 ++--
 4 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
index 38dccdddfaaa..89b915a3d6e2 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
@@ -166,7 +166,7 @@
 			logo,kernel = "logo_kernel.bmp";
 			logo,mode = "center";
 			charge_logo,mode = "center";
-			connect = <&vopl_out_hdmi>;
+			connect = <&vopb_out_hdmi>;
 		};
 
 		route_dsi: route-dsi {
@@ -175,7 +175,7 @@
 			logo,kernel = "logo_kernel.bmp";
 			logo,mode = "center";
 			charge_logo,mode = "center";
-			connect = <&vopb_out_dsi>;
+			connect = <&vopl_out_dsi>;
 		};
 
 		route_edp: route-edp {
@@ -184,7 +184,7 @@
 			logo,kernel = "logo_kernel.bmp";
 			logo,mode = "center";
 			charge_logo,mode = "center";
-			connect = <&vopb_out_edp>;
+			connect = <&vopl_out_edp>;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
index 82f34590ce03..0be01405668a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
@@ -296,6 +296,10 @@
 	};
 };
 
+&edp_in_vopb {
+	status = "disabled";
+};
+
 &hdmi {
 	pinctrl-names = "default";
 	pinctrl-0 = <&hdmi_i2c_xfer>, <&hdmi_cec>;
@@ -305,6 +309,10 @@
 	status = "okay";
 };
 
+&hdmi_in_vopl {
+	status = "disabled";
+};
+
 &i2c1 {
 	status = "okay";
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
index 78c2d8521419..3d3b2505b9b6 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
@@ -286,7 +286,7 @@
 	};
 };
 
-&edp_in_vopl {
+&edp_in_vopb {
 	status = "disabled";
 };
 
@@ -329,7 +329,7 @@
 		<000000000 0x0000 0x0000 0x0000>;
 };
 
-&hdmi_in_vopb {
+&hdmi_in_vopl {
 	status = "disabled";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
index ab985040c8a9..ad8de3a08963 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
@@ -291,7 +291,7 @@
 	};
 };
 
-&edp_in_vopl {
+&edp_in_vopb {
 	status = "disabled";
 };
 
@@ -334,7 +334,7 @@
 		<000000000 0x0000 0x0000 0x0000>;
 };
 
-&hdmi_in_vopb {
+&hdmi_in_vopl {
 	status = "disabled";
 };
 
-- 
2.35.3

